src/pages/checkOutReport.js

```diff
@@ -0,0 +1,99 @@
+import fieldToConceptIdMapping from "../fieldToConceptIdMapping.js";
+
+
+export const checkOutReportTemplate = (auth, route) => {
+    auth.onAuthStateChanged(async user => {
+        if(user){
+            const response = await userAuthorization(route, user.displayName ? user.displayName : user.email);
+            if ( response.isBiospecimenUser === false ) {
+                document.getElementById("contentBody").innerHTML = "Authorization failed you lack permissions to use this dashboard!";
+                document.getElementById("navbarNavAltMarkup").innerHTML = unAuthorizedUser();
+                return;
+            }
+            if(!response.role) return;
+            renderCheckOutReport();
+        }
+        else {
+            document.getElementById('navbarNavAltMarkup').innerHTML = homeNavBar();
+            window.location.hash = '#';
+        }
+    });
+}
+
+
+export const renderCheckOutReport = async () => {  
+    let template = `
+            <div class="container">
+            <div class="row">
+                <div class="col-lg-2" style="margin-bottom:20px">
+                    <h2>Reports</h2>
+                    ${reportSideNavBar()}
+                </div>
+                <div class="col-lg-10">
+                    <div class="row">
+                        <div class="col-lg">
+                            <table id="populateCheckOutTable" class="table table-bordered">
+                            </table>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>`;
+    document.getElementById('contentBody').innerHTML = template;
+    removeActiveClass('nav-link', 'active');
+    const navBarBtn = document.getElementById('navBarCheckoutReport');
+    navBarBtn.classList.add('active');
+    populateCheckOutTable();
+}
+
+export const populateCheckOutTable = async () => {
+    const currTable = document.getElementById('populateCheckOutTable');
+    currTable.innerHTML = '';
+    
+    const headerRow = currTable.insertRow();
+    headerRow.innerHTML = `
+      <th><b>Connect ID</b></th>
+      <th><b>Last Name</b></th>
+      <th><b>First Name</b></th>
+      <th><b>Check-In Date/Time</b></th>
+      <th><b>Go to Check-Out</b></th>
+    `;
+    
+    showAnimation();
+    const data = await findParticipant(`checkedIn=${true}`).then(res => res.data);
+    
+    for (const item of data) {
+      if (!item[fieldToConceptIdMapping.collection.selectedVisit][fieldToConceptIdMapping.baseline.visitId][
+          fieldToConceptIdMapping.checkOutDateTime]) {
+        const newRow = currTable.insertRow();
+        newRow.innerHTML = `
+          <td>${item['Connect_ID']}</td>
+          <td>${item[fieldToConceptIdMapping.lName]}</td>
+          <td>${item[fieldToConceptIdMapping.fName]}</td>
+          <td>${convertISODateTime(item[fieldToConceptIdMapping.collection.selectedVisit][fieldToConceptIdMapping.baseline.visitId]
```

Comment:
Similar to the non existent first name and last name comment above. the participant 
```suggestion
          <td>${convertISODateTime(item[fieldToConceptIdMapping.collection.selectedVisit]?.[fieldToConceptIdMapping.baseline.visitId]
```

Reply:
got it.